multiversx_sdk_cli/cli_wallet.py

```diff
@@ -89,38 +111,155 @@ def setup_parser(args: List[str], subparsers: Any) -> Any:
+        format = "pem"
+    if args.output_path:
+        logger.warning("The --output-path option is deprecated, use --outfile instead.")
+        outfile = args.output_path
+
     mnemonic = Mnemonic.generate()
     print(f"Mnemonic: {mnemonic.get_text()}")
-    secret_key = mnemonic.derive_key()
-    pubkey = secret_key.generate_public_key()
-    address = Address(pubkey.hex())
-    output_path = args.output_path
 
-    if args.pem:
+    if format is None:
+        return
+    if outfile is None:
+        raise KnownError("The --outfile option is required when --format is specified.")
+
+    outfile = Path(outfile).expanduser().resolve()
+    if outfile.exists():
+        raise KnownError(f"File already exists, will not overwrite: {outfile}")
+
+    if format == WALLET_FORMAT_RAW_MNEMONIC:
+        outfile.write_text(mnemonic.get_text())
+    elif format == WALLET_FORMAT_KEYSTORE_MNEMONIC:
+        password = getpass.getpass("Enter a new password (for keystore):")
+        wallet = UserWallet.from_mnemonic(mnemonic.get_text(), password)
+        wallet.save(outfile)
+    elif format == WALLET_FORMAT_KEYSTORE_SECRET_KEY:
+        password = getpass.getpass("Enter a new password (for keystore):")
+        secret_key = mnemonic.derive_key()
+        wallet = UserWallet.from_secret_key(secret_key, password)
+        wallet.save(outfile, address_hrp)
+    elif format == WALLET_FORMAT_PEM:
+        secret_key = mnemonic.derive_key()
+        pubkey = secret_key.generate_public_key()
+        address = pubkey.to_address(address_hrp)
         pem_file = UserPEM(address.bech32(), secret_key)
-        pem_file_path = prepare_file(args.output_path, ".pem")
-        pem_file.save(pem_file_path)
-        logger.info(f"Pem wallet generated: {pem_file_path}")
-    if args.json:
-        json_file = prepare_file(args.output_path, ".json")
-        password = getpass.getpass("Enter a new password:")
+        pem_file.save(outfile)
+    else:
+        raise KnownError(f"Unknown format: {format}")
+
+    logger.info(f"Wallet ({format}) saved: {outfile}")
+
 
-        keyfile = convert_to_keyfile_object(secret_key.buffer, pubkey.buffer, password, None, "erd")
-        with open(output_path, 'w') as json_file:
-            dump(keyfile, json_file, indent=4)
+def convert_wallet(args: Any):
+    infile = Path(args.infile).expanduser().resolve() if args.infile else None
+    outfile = Path(args.outfile).expanduser().resolve() if args.outfile else None
+    in_format = args.in_format
+    out_format = args.out_format
+    address_index = args.address_index
+    address_hrp = args.address_hrp
 
-        logger.info(f"Json wallet generated: {json_file}")
+    if outfile and outfile.exists():
+        raise KnownError(f"File already exists, will not overwrite: {outfile}")
+
+    if infile:
+        input_text = infile.read_text()
+    else:
+        print(f"Press 'Ctrl-D' (Linux / MacOS) or 'Ctrl-Z' (Windows) when done.")
```

Comment:
as discussed, might prepend a `Insert text` to this message

Reply:
Fixed.